### PR TITLE
Update cachecontrol library

### PR DESCRIPTION
**Pull request by soyuka**
https://github.com/pquerna/cachecontrol/pull/18 got merged and the code needed an update accordingly
Also adds support for the no-cache directive, note that validation hasn't been done yet.

### CODE REVIEW

File: httpcache.go [hunk 1]
```diff
@@ -169,21 +169,24 @@ func (c *Cache) writeResponse(w http.ResponseWriter, rdr io.Reader, fromCache bo
...
+	obj := cacheobject.Object{ ReqMethod: r.Method, ReqDirectives: reqDir }
+	rv := cacheobject.ObjectResults{}
+	cacheobject.CachableRequestObject(&obj, &rv)
+
+	if rv.OutErr != nil || len(rv.OutReasons) > 0 {
+		w.Header().Add("Cache-Status", fmt.Sprintf(userAgent+`; fwd=request; detail="%v"`, rv.OutReasons))
+		return next.ServeHTTP(w, r)
+	}
+
+	if reqDir.NoCache {
```
Comment (dunglas) [inline httpcache.go:188]:
```suggestion
        // TODO: implement no-cache properly (add support for validation)
	if reqDir.NoCache {
```

File: httpcache_test.go [hunk 2]
```diff
@@ -128,3 +128,24 @@ func TestOlricConfig(t *testing.T) {
 
 	tester.AssertGetResponse(`http://localhost:9080/cache-max-age`, 200, "Hello, max-age!")
 }
+
+func TestNoCache(t *testing.T) {
```
Comment (dunglas) [inline httpcache_test.go:132]: Could you also add a test with `no-cache` set on the request?

File: httpcache.go [hunk 3]
```diff
@@ -169,21 +169,25 @@ func (c *Cache) writeResponse(w http.ResponseWriter, rdr io.Reader, fromCache bo
...
-	case http.MethodGet:
-	case http.MethodHead:
-	case http.MethodPost:
-	default:
-		// method not cacheable
-		w.Header().Add("Cache-Status", userAgent+"; fwd=request; detail=METHOD")
+	reqDir, err := cacheobject.ParseRequestCacheControl(r.Header.Get("Cache-Control"))
+
+	if err != nil {
+		w.Header().Add("Cache-Status", userAgent+"; fwd=request; detail=MALFORMED-CACHE-CONTROL")
```
[the remark this answers is not in the document]
Comment (soyuka) [reply httpcache.go:175]: I think that we use `Sprintf` only when there's more then one string concatenation

**Comment by darkweak** [comment]
We close this PR as we discussed on slack.

**Pull request closed without merging**